src/vs/workbench/contrib/terminal/browser/terminalConfigHelper.ts

```diff
@@ -208,11 +212,25 @@ export class TerminalConfigHelper implements IBrowserTerminalConfigHelper {
 
 	public setWorkspaceShellAllowed(isAllowed: boolean): void {
 		this._onWorkspacePermissionsChanged.fire(isAllowed);
-		this._storageService.store(IS_WORKSPACE_SHELL_ALLOWED_STORAGE_KEY, isAllowed, StorageScope.WORKSPACE);
+		const activeWorkspaceRootUri = this._historyService.getLastActiveWorkspaceRoot();
+		const lastActiveWorkspace = activeWorkspaceRootUri ? this._workspaceContextService.getWorkspaceFolder(activeWorkspaceRootUri) : undefined;
+		if (lastActiveWorkspace) {
+			const folderUri = IS_WORKSPACE_SHELL_ALLOWED_STORAGE_KEY + '_' + lastActiveWorkspace.uri.path;
```

Comment:
This will introduce a new key that is prioritized over the old one, so everyone that's already answered the prompt will see it again. Can we check the workspace scope first and fall back to this to prevent that?